designs/umbra/src/front.mjs

```diff
@@ -0,0 +1,226 @@
+import { base } from './base.mjs'
+
+export const front = {
+  name: 'umbra.front',
+  from: base,
+  draft: draftUmbraFront,
+}
+
+/*
+ * This drafts the front of Umbra, or rather recycles what's needed from base
+ */
+function draftUmbraFront({
+  Point,
+  Path,
+  points,
+  paths,
+  store,
+  sa,
+  Snippet,
+  snippets,
+  options,
+  expand,
+  macro,
+  part,
+}) {
+  if (store.get('bulge')) {
+    paths.seamBase = new Path()
+      .move(points.cfBulgeSplit)
+      .curve(points.bulgeCpBottom, points.cfMiddleBulgeCp, points.cfMiddleBulge)
+      .line(points.cfBackGussetBulge)
+  } else {
+    paths.seamBase = new Path().move(points.cfBackGussetBulge)
+  }
+  paths.seamBase = paths.seamBase
+    .line(points.backGussetSplitBulge)
+    .join(paths.elasticLegFront)
+    .line(points.sideWaistbandFront)
+    ._curve(points.cfWaistbandDipCpFront, points.cfWaistbandDipFront)
+    .hide()
+  if (expand) {
+    macro('mirror', {
+      mirror: [new Point(0, 0), new Point(0, 100)],
+      paths: ['seamBase'],
+      clone: true,
+    })
+    paths.seam = paths.seamBase.join(paths.mirroredSeamBase.reverse())
+    paths.mirroredSeamBase.hide()
+
+    macro('mirror', {
+      mirror: [new Point(0, 0), new Point(0, 100)],
+      paths: ['pocketShape', 'pocketHem', 'pocketHem', 'zipper', 'zipperCut'],
+      points: ['pocketTop'],
+      clone: true,
+    })
+  } else {
+    paths.seam = paths.seamBase.clone().close()
+  }
+  paths.seam.unhide().addClass('fabric')
+  if (sa) {
+    paths.saBase = new Path()
+      .move(points.cfBackGussetBulge)
+      .line(points.backGussetSplitBulge)
+      .join(paths.elasticLegFront)
+      .line(points.sideWaistbandFront)
+      ._curve(points.cfWaistbandDipCpFront, points.cfWaistbandDipFront)
+      .hide()
+      .hide()
+
+    if (expand) {
+      macro('mirror', {
+        mirror: [new Point(0, 0), new Point(0, 100)],
+        paths: ['saBase'],
+        clone: true,
+      })
+      paths.saBase = paths.saBase.join(paths.mirroredSaBase.reverse()).close().hide()
+      paths.sa = paths.saBase.offset(sa).setClass('fabric sa').unhide()
+    } else {
+      paths.sa = paths.saBase
+        .offset(sa)
+        .reverse()
+        .line(new Point(0, points.cfBackGussetBulge.y))
+        .line(points.cfBulgeSplit)
+        .reverse()
+        .line(points.cfWaistbandDipFront)
+        .setClass('fabric sa')
+        .unhide()
+    }
+  }
+  store.cutlist.setCut({ cut: 1, from: 'fabric', onFold: !expand })
+  store.cutlist.addCut({ cut: 1, from: 'lining', onFold: !expand })
+  points.title = points.cfWaistbandDipFront
+    .shiftFractionTowards(points.sideLegFront, 0.14)
+    .shiftFractionTowards(points.cfMiddle, 0.25)
+  if (!expand) {
```

Comment:
This is perhaps a nitpick, but I find:

```mjs
if (something) {
  // ...
} 
else {
  // ...
}
```

more intutitive than:

```mjs
if (!something) { 
  // ...
} 
else {
  // ...
}
```

Reply:
You're right